src/module/account/settings/_component/FormInput/FormInput.tsx

```diff
@@ -0,0 +1,72 @@
+'use client';
+
+import React, { useState } from 'react';
+import {
+  FormFieldButton,
+  FormGroup,
+  FormGroupField,
+  FormGroupLabel,
+} from '@/shared/components/form/FormElements';
+import Error from '@/shared/components/form/Error';
+import styled from 'styled-components';
+import { FieldErrors } from 'react-hook-form';
+import EyeIcon from 'mdi-react/EyeIcon';
+import FormIcon from './_component/FormIcon';
+
+interface FormInputProps {
+  placeholder: string;
+  errors: FieldErrors;
+  field: { [key: string]: any };
+  isAboveError?: boolean;
+  disabled?: boolean;
+}
+
+function FormInput({ field, placeholder, errors, isAboveError, disabled }: FormInputProps) {
+  const [passwordShown, setPasswordShown] = useState(false);
+  const { name } = field;
+  return (
+    <FormGroup>
+      <FormGroupLabel>{placeholder}</FormGroupLabel>
+      <FormGroupField>
+        <FormIcon name={name} />
+        {name === 'password' && (
```

Comment:
Best to put them as separate components instead of switching them based on `name`

Reply:
Okkkkk